### PR TITLE
[stable/postgresql] Allow connection details to be set as global

### PR DESCRIPTION
**Pull request by javsalgar**
Signed-off-by: Javier J. Salmeron Garcia <jsalmeron@bitnami.com>

#### What this PR does / why we need it:
Extracted from the README

In more complex scenarios, we may have the following tree of dependencies

 ```
                     +--------------+
                     |              |
        +------------+   Chart 1    +-----------+
        |            |              |           |
        |            --------+------+           |
        |                    |                  |
        |                    |                  |
        |                    |                  |
        |                    |                  |
        v                    v                  v
+-------+------+    +--------+------+  +--------+------+
|              |    |               |  |               |
|  PostgreSQL  |    |  Sub-chart 1  |  |  Sub-chart 2  |
|              |    |               |  |               |
+--------------+    +---------------+  +---------------+
```

 The three charts below depend on the parent chart Chart 1. However, subcharts 1 and 2 may need to connect to PostgreSQL as well. In order to do so, subcharts 1 and 2 need to know the PostgreSQL credentials, so one option for deploying could be:

 ```
helm install chart1 --set postgresql.postgresqlPassword=testtest --set subchart1.postgresql.postgresqlPassword=testtest --set subchart2.postgresql.postgresqlPassword=testtest --set postgresql.postgresqlDatabase=db1 --set subchart1.postgresql.postgresqlDatabase=db1 --set subchart2.postgresql.postgresqlDatabase=db1
```

 If the number of dependent sub-charts increases, executing `helm install` can become increasingly difficult. An alternative would be to set the credentials using global variables as follows:

 ```
helm install chart1 --set global.postgresql.postgresqlPassword=testtest --set global.postgresql.postgresqlDatabase=db1
```

 This way, the credentials will be available in all of the subcharts.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Review by juan131 — COMMENTED** [review]
I'm fine with this approach for this really complex scenarios but I'd like to hear from other maintainers about their opinion.

cc/ @desaintmartin @tompizmor @prydonius

**Comment by javsalgar** [comment]
Ping @desaintmartin @prydonius @tompizmor

**Comment by desaintmartin** [comment]
Is there some common consensus about this in the helm community / this repository? It would be counterproductive if some other chart starts to use another solution.

If think it deserves a discussion during the meeting.

**Comment by carrodher** [comment]
Yes, I agree, we can add this discussion to the following "Charts Office Hours" meeting. @javsalgar can you add it to the [agenda](https://docs.google.com/document/d/1h6UTTuNRbFI81higrN3JUV2XxyzqqVjZET4Xz4WTR-8/edit) if it's okay to you?

**Comment by WarheadsSE** [comment]
Thank you for spawning this discussion @javsalgar :bowing_man:

**Review by mattfarina — APPROVED** [review]

**Comment by mattfarina** [comment]
This looks ok to me. I'd like @prydonius to take a pass at it and it needs a rebase from master.

**Comment by javsalgar** [comment]
Thanks @mattfarina , merge issues fixed

File: stable/postgresql/templates/_helpers.tpl [hunk 1]
```diff
@@ -74,6 +74,133 @@ Also, we can't use a single if because lazy evaluation is not an option
 {{- end -}}
 {{- end -}}
 
+{{/*
+Return PostgreSQL password
+*/}}
+{{- define "postgresql.password" -}}
+{{- if .Values.global }}
```
Comment (prydonius) [inline stable/postgresql/templates/_helpers.tpl]: you might be able to reduce the amount of nested if/else cases you need if you set a default for global inside values.yaml that can help ensure that `global.postgresql` is defined:

```
global:
 postgresql: {}
```

**Review by prydonius — APPROVED** [review]
mostly lgtm

**Review by prydonius — APPROVED** [review]
nice, thanks!

File: stable/postgresql/templates/_helpers.tpl (hunk 1 above)
Comment (javsalgar) [reply stable/postgresql/templates/_helpers.tpl]: Done. Thanks!

**Pull request merged**